src/sidebar/components/test/markdown-editor-test.js

```diff
@@ -301,6 +305,303 @@ describe('MarkdownEditor', () => {
     assert.equal(inputField.prop('aria-label'), 'Annotation body');
   });
 
+  describe('keyboard navigation', () => {
+    let newContainer;
+    let wrapper;
+
+    beforeEach(() => {
+      newContainer = document.createElement('div');
+      document.body.appendChild(newContainer);
+      wrapper = mount(<MarkdownEditor label="Test editor" text="test" />, {
+        attachTo: newContainer,
+      });
+    });
+
+    afterEach(() => {
+      newContainer.remove();
+    });
+
+    const arrowRight = () => {
+      wrapper
+        .find('.markdown-editor__toolbar')
+        .simulate('keydown', { key: 'ArrowRight' });
+    };
+    const arrowLeft = () => {
+      wrapper
+        .find('.markdown-editor__toolbar')
+        .simulate('keydown', { key: 'ArrowLeft' });
+    };
+    const end = () => {
+      wrapper
+        .find('.markdown-editor__toolbar')
+        .simulate('keydown', { key: 'End' });
+    };
+    const home = () => {
+      wrapper
+        .find('.markdown-editor__toolbar')
+        .simulate('keydown', { key: 'Home' });
+    };
+
+    const testRovingIndex = () => {
+      assert.isTrue(document.activeElement.getAttribute('tabIndex') === '0');
+      // There should only be one "0" `tabIndex` value at a time which
+      // should be set on the focused element. All other `tabIndex` values
+      // on elements shall be "-1".
+      assert.equal(
+        wrapper.find('ToolbarButton[tabIndex="0"]').length +
+          wrapper.find('a[tabIndex="0"]').length,
+        1
+      );
+    };
+
+    context('when `isPreviewing` is false', () => {
+      it('changes focus to the second button when pressing `right` arrow', () => {
```

Comment:
There is quite a lot of repetition in these tests. Perhaps some of this could be factored out to make it easier to follow at a high level. For example, extracting a helper for checking if the focused button's label matches some string.

Reply:
I'll update this PR again later tonight, I don't want to delay anymore on this, but I think you're right. I did remove a few tests I felt were un-needed but added 2 more than seemed logically appropriate. I also removed the `commanFn` indirection as it was the same for each test in the set.